test(transformer/using): alter test to demonstrate semantic mismatch

overlookmotel opened the pull request:
Where a `for (using x of ...)` statement has a single statement as its body, `enter_for_of_statement` wraps that statement in a `BlockStatement`. This introduces a new scope. We are missing logic to re-parent any scopes within that statement to be children of the new block scope.

This PR alters one of the tests to demonstrate this problem.

overlookmotel commented:
* **#9700** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9700?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9700?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9700).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
@camc314 I think using `TraverseCtx::insert_scope_below_statement` to create the scope for the `BlockStatement` should fix this one.

camc314 commented:
### Merge activity

* **Mar 13, 7:44 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Mar 13, 7:44 AM EDT**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
test(transformer/using): alter test to demonstrate semantic mismatch (#9700)

Where a `for (using x of ...)` statement has a single statement as its body, `enter_for_of_statement` wraps that statement in a `BlockStatement`. This introduces a new scope. We are missing logic to re-parent any scopes within that statement to be children of the new block scope.

This PR alters one of the tests to demonstrate this problem.
